cmd/bnsd/x/username/username.go

```diff
@@ -49,25 +50,30 @@ func (u Username) String() string {
 }
 
 func (u Username) Validate() error {
-	if !validUsername(string(u)) {
-		return errors.Wrap(errors.ErrInput, "invalid username")
+	const usernameSeparator = `*`
+	chunks := strings.SplitN(string(u), usernameSeparator, 2)
+	if len(chunks) != 2 {
+		return errors.Wrap(errors.ErrInput, "missing separator")
 	}
 
+	var errs error
+	if !validName(chunks[0]) {
+		errs = errors.AppendField(errs, "Name", errors.ErrInput)
+	}
+	if !validNamespace(chunks[1]) {
+		errs = errors.AppendField(errs, "Namespace", errors.ErrInput)
+	}
+	return errs
+}
+
+var (
+	validName = regexp.MustCompile(`^[a-z0-9\-\._]{0,64}$`).MatchString
```

Comment:
you may want to use `"^[a-z0-9\\-\\._]{0,64}$"` instead

Reply:
Can you explain more why would I use your pattern instead of what there is currently?

Reply:
Inside of the characters class, the escaping should not be necessray at all. See https://regex101.com/r/kSHpA0/1

Reply:
My ide was complaining about redundant character escaping. But it should be ``^[a-z0-9\-._]{0,64}$`` instead. anyway, it works.